packages/google_maps_flutter/google_maps_flutter_ios/example/ios/RunnerTests/GoogleMapsTests.m

```diff
@@ -39,4 +44,15 @@ - (void)testFrameObserver {
   XCTAssertEqual(mapView.frameObserverCount, 0);
 }
 
+- (void)testMapsServiceSync {
+  id registrar = OCMProtocolMock(@protocol(FlutterPluginRegistrar));
+  FLTGoogleMapFactory *factory1 = [[FLTGoogleMapFactory alloc] initWithRegistrar:registrar];
+  XCTAssertNotNil(factory1.sharedMapServices);
+  FLTGoogleMapFactory *factory2 = [[FLTGoogleMapFactory alloc] initWithRegistrar:registrar];
+  // Test pointer equality, should be same retained singleton +[GMSServices sharedServices] object.
+  // Retaining the opaque object should be enough to avoid multiple internal initializations,
+  // but don't test the internals of the GoogleMaps API. Assume that it does what is documented.
```

Comment:
optional nits: add a link to the doc where this optimization is explain: https://developers.google.com/maps/documentation/ios-sdk/reference/interface_g_m_s_services#a436e03c32b1c0be74e072310a7158831

Reply:
Done.